Plugin doesn't work on Windows

**Is this a request for help?**:
Yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Environment (plugin version, maven version, OS, ...)**:
Plugin version 4.11, Windows 10, Maven 3.6.1

**What happened**:
Plugin fails to extract the helm windows binary which is a zip file and not tar.gz.

**What you expected to happen**:
The plugin to download the windows binary and execute it.

Stacktrace from maven execution:
`org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.kiwigrid:helm-maven-plugin:4.11:init (package-helm-chart) on project fsservice-server: Unable to download and extract helm executable.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Unable to download and extract helm executable.
    at com.kiwigrid.helm.maven.plugin.InitMojo.downloadAndUnpackHelm (InitMojo.java:119)
    at com.kiwigrid.helm.maven.plugin.InitMojo.execute (InitMojo.java:61)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.util.zip.ZipException: Not in GZIP format
    at java.util.zip.GZIPInputStream.readHeader (GZIPInputStream.java:165)
    at java.util.zip.GZIPInputStream.<init> (GZIPInputStream.java:79)
    at java.util.zip.GZIPInputStream.<init> (GZIPInputStream.java:91)
    at com.kiwigrid.helm.maven.plugin.InitMojo.downloadAndUnpackHelm (InitMojo.java:92)
    at com.kiwigrid.helm.maven.plugin.InitMojo.execute (InitMojo.java:61)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)`

--- a/src/main/java/com/kiwigrid/helm/maven/plugin/InitMojo.java
+++ b/src/main/java/com/kiwigrid/helm/maven/plugin/InitMojo.java
@@ -51,7 +51,7 @@ public class InitMojo extends AbstractHelmMojo {
 	private boolean skipInit;
 
 	public void execute() throws MojoExecutionException {
-		
+
 		if (skip || skipInit) {
 			getLog().info("Skip init");
 			return;
@@ -99,7 +99,7 @@ public class InitMojo extends AbstractHelmMojo {
 			getLog().info("Found helm executable, skip init.");
 			return;
 		}
-		
+
 		getLog().info("Downloading Helm ...");
 		boolean found = false;
 		try (InputStream dis = new URL(getHelmDownloadUrl()).openStream();
@@ -144,7 +144,7 @@ public class InitMojo extends AbstractHelmMojo {
 
 	private void addExecPermission(final Path helm) throws IOException {
 		Set<String> fileAttributeView = FileSystems.getDefault().supportedFileAttributeViews();
-		
+
 		if (fileAttributeView.contains("posix")) {
 			final Set<PosixFilePermission> permissions;
 			try {
@@ -155,7 +155,7 @@ public class InitMojo extends AbstractHelmMojo {
 			}
 			permissions.add(PosixFilePermission.OWNER_EXECUTE);
 			Files.setPosixFilePermissions(helm, permissions);
-			
+
 		} else if (fileAttributeView.contains("acl")) {
 			String username = System.getProperty("user.name");
 			UserPrincipal userPrincipal = FileSystems.getDefault().getUserPrincipalLookupService().lookupPrincipalByName(username);
@@ -187,7 +187,7 @@ public class InitMojo extends AbstractHelmMojo {
 		String cmd = getHelmExecuteablePath()
 				+ " init --client-only" + (skipRefresh ? " --skip-refresh" : "")
 				+ (StringUtils.isNotEmpty(getHelmHomeDirectory()) ? " --home=" + getHelmHomeDirectory() : "");
-		
+
 		getLog().info("Running: " + cmd);
 		callCli(cmd,
 				"Unable to call helm init",
@@ -222,7 +222,7 @@ public class InitMojo extends AbstractHelmMojo {
 		} catch (CompressorException e) {
 			getLog().debug("Unknown type of compressed stream", e);
 		}
-		
+
 		// If compressed then wrap with compressor stream
 		if (compressorType != null) {
 			try {

--- a/src/main/java/com/kiwigrid/helm/maven/plugin/InitMojo.java
+++ b/src/main/java/com/kiwigrid/helm/maven/plugin/InitMojo.java
@@ -1,19 +1,31 @@
 package com.kiwigrid.helm.maven.plugin;
 
-import java.io.File;
+import java.io.BufferedInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.PasswordAuthentication;
 import java.net.URL;
+import java.nio.file.FileSystems;
 import java.nio.file.Files;
+import java.nio.file.LinkOption;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.attribute.AclEntry;
+import java.nio.file.attribute.AclEntryPermission;
+import java.nio.file.attribute.AclEntryType;
+import java.nio.file.attribute.AclFileAttributeView;
 import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.UserPrincipal;
+import java.util.List;
 import java.util.Set;
-import java.util.zip.GZIPInputStream;
 
-import com.kiwigrid.helm.maven.plugin.pojo.HelmRepository;
-import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveStreamFactory;
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
 import org.apache.commons.compress.utils.IOUtils;
 import org.apache.commons.lang3.SystemUtils;
 import org.apache.maven.plugin.MojoExecutionException;
@@ -21,6 +33,8 @@ import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.codehaus.plexus.util.StringUtils;
+
+import com.kiwigrid.helm.maven.plugin.pojo.HelmRepository;
 
 /**
  * Mojo for initializing helm
@@ -85,20 +99,22 @@ public class InitMojo extends AbstractHelmMojo {
 			getLog().info("Found helm executable, skip init.");
 			return;
 		}
-
+		
 		getLog().info("Downloading Helm ...");
 		boolean found = false;
-		try (TarArchiveInputStream is = new TarArchiveInputStream(
-				new GZIPInputStream(new URL(getHelmDownloadUrl()).openStream()))) {
+		try (InputStream dis = new URL(getHelmDownloadUrl()).openStream();
+			 InputStream cis = createCompressorInputStream(dis);
+			 ArchiveInputStream is = createArchiverInputStream(cis)) {
 
 			// create directory if not present
 			Files.createDirectories(directory);
 
 			// get helm executable entry
-			while (is.getNextEntry() != null) {
+			ArchiveEntry entry = null;
+			while ((entry = is.getNextEntry()) != null) {
 
-				String name = is.getCurrentEntry().getName();
-				if (is.getCurrentEntry().isDirectory() || (!name.endsWith("helm.exe") && !name.endsWith("helm"))) {
+				String name = entry.getName();
+				if (entry.isDirectory() || (!name.endsWith("helm.exe") && !name.endsWith("helm"))) {
 					getLog().debug("Skip archive entry with name: " + name);
 					continue;
 				}
@@ -117,7 +133,8 @@ public class InitMojo extends AbstractHelmMojo {
 
 		} catch (IOException e) {
 			throw new MojoExecutionException("Unable to download and extract helm executable.", e);
-		}
+		} 
+
 		if (!found) {
 			throw new MojoExecutionException("Unable to find helm executable in tar file.");
 		}
@@ -126,15 +143,29 @@ public class InitMojo extends AbstractHelmMojo {
 	}
 
 	private void addExecPermission(final Path helm) throws IOException {
-		final Set<PosixFilePermission> permissions;
-		try {
-			permissions = Files.getPosixFilePermissions(helm);
-		} catch (UnsupportedOperationException e) {
-			getLog().debug("Exec file permission is not set", e);
-			return;
+		Set<String> fileAttributeView = FileSystems.getDefault().supportedFileAttributeViews();
+		
+		if (fileAttributeView.contains("posix")) {
+			final Set<PosixFilePermission> permissions;
+			try {
+				permissions = Files.getPosixFilePermissions(helm);
+			} catch (UnsupportedOperationException e) {
+				getLog().debug("Exec file permission is not set", e);
+				return;
+			}
+			permissions.add(PosixFilePermission.OWNER_EXECUTE);
+			Files.setPosixFilePermissions(helm, permissions);
+			
+		} else if (fileAttributeView.contains("acl")) {
+			String username = System.getProperty("user.name");
+			UserPrincipal userPrincipal = FileSystems.getDefault().getUserPrincipalLookupService().lookupPrincipalByName(username);
+			AclEntry aclEntry = AclEntry.newBuilder().setPermissions(AclEntryPermission.EXECUTE).setType(AclEntryType.ALLOW).setPrincipal(userPrincipal).build();
+
+			AclFileAttributeView acl = Files.getFileAttributeView(helm, AclFileAttributeView.class, LinkOption.NOFOLLOW_LINKS);
+			List<AclEntry> aclEntries = acl.getAcl();
+			aclEntries.add(aclEntry);
+			acl.setAcl(aclEntries);
 		}
-		permissions.add(PosixFilePermission.OWNER_EXECUTE);
-		Files.setPosixFilePermissions(helm, permissions);
 	}
 
 	private void verifyLocalHelmBinary() throws MojoExecutionException {
@@ -153,10 +184,55 @@ public class InitMojo extends AbstractHelmMojo {
 	private void initHelmClient() throws MojoExecutionException {
 		getLog().info("Run helm init...");
 
-		callCli(getHelmExecuteablePath()
-						+ " init --client-only" + (skipRefresh ? " --skip-refresh" : "")
-						+ (StringUtils.isNotEmpty(getHelmHomeDirectory()) ? " --home=" + getHelmHomeDirectory() : ""),
+		String cmd = getHelmExecuteablePath()
+				+ " init --client-only" + (skipRefresh ? " --skip-refresh" : "")
+				+ (StringUtils.isNotEmpty(getHelmHomeDirectory()) ? " --home=" + getHelmHomeDirectory() : "");
+		
+		getLog().info("Running: " + cmd);
+		callCli(cmd,
 				"Unable to call helm init",
 				false);
+	}
+
+	private ArchiveInputStream createArchiverInputStream(InputStream is) throws MojoExecutionException {
+		// Stream must support mark to allow for auto detection of archiver
+		if (!is.markSupported()) {
+			is = new BufferedInputStream(is);
+		}
+
+		try {
+			ArchiveStreamFactory archiveStreamFactory = new ArchiveStreamFactory();
+			return archiveStreamFactory.createArchiveInputStream(is);
+
+		} catch (ArchiveException e) {
+			throw new MojoExecutionException("Unsupported archive type downloaded", e);
+		}
+	}
+
+	private InputStream createCompressorInputStream(InputStream is) throws MojoExecutionException {
+		// Stream must support mark to allow for auto detection of compressor
+		if (!is.markSupported()) {
+			is = new BufferedInputStream(is);
+		}
+
+		// Detect if stream is compressed
+		String compressorType = null;
+		try {
+			compressorType = CompressorStreamFactory.detect(is);
+		} catch (CompressorException e) {
+			getLog().debug("Unknown type of compressed stream", e);
+		}
+		
+		// If compressed then wrap with compressor stream
+		if (compressorType != null) {
+			try {
+				CompressorStreamFactory compressorFactory = new CompressorStreamFactory();
+				return compressorFactory.createCompressorInputStream(compressorType, is);
+			} catch (CompressorException e) {
+				throw new MojoExecutionException("Unsupported compressor type: " + compressorType);
+			}
+		}
+
+		return is;
 	}
 }

--- a/src/test/java/com/kiwigrid/helm/maven/plugin/AbstractHelmMojoTest.java
+++ b/src/test/java/com/kiwigrid/helm/maven/plugin/AbstractHelmMojoTest.java
@@ -1,20 +1,5 @@
 package com.kiwigrid.helm.maven.plugin;
 
-import org.apache.commons.lang3.SystemUtils;
-import org.apache.maven.plugin.MojoExecutionException;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Nested;
-import org.junit.jupiter.api.Test;
-import org.mockito.Mockito;
-
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.List;
-
 import static java.nio.file.Files.write;
 import static java.util.Arrays.asList;
 import static org.apache.commons.io.FileUtils.deleteQuietly;
@@ -24,6 +9,20 @@ import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.doReturn;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 
 class AbstractHelmMojoTest {
 
@@ -74,7 +73,7 @@ class AbstractHelmMojoTest {
     void getChartDirectoriesReturnChartDirectoriesWithAntPatternsExclusion() throws MojoExecutionException {
         Path baseChartsDirectory = getBaseChartsDirectory();
 
-        subjectSpy.setExcludes(new String[]{"**/exclude*"});
+        subjectSpy.setExcludes(new String[]{"**" + File.separator + "exclude*"});
 
         List<String> chartDirectories = subjectSpy.getChartDirectories(baseChartsDirectory.toString());
 
@@ -146,7 +145,7 @@ class AbstractHelmMojoTest {
     private Path addHelmToTestPath() throws IOException { return write(testHelmExecutablePath, new byte[]{}); }
 
     private Path getBaseChartsDirectory() {
-        return Paths.get(this.getClass().getResource("Chart.yaml").getFile()).getParent();
+    	return new File(getClass().getResource("Chart.yaml").getFile()).toPath().getParent();
     }
 
     @AfterEach

--- a/src/test/java/com/kiwigrid/helm/maven/plugin/InitMojoTest.java
+++ b/src/test/java/com/kiwigrid/helm/maven/plugin/InitMojoTest.java
@@ -123,7 +123,7 @@ public class InitMojoTest {
 	}
 	
 	private String getOsSpecificDownloadURL() {
-		return getOsSpecificDownloadURL(Os.OS_FAMILY);
+		return getOsSpecificDownloadURL(Os.OS_FAMILY == Os.FAMILY_UNIX ? "linux" : Os.OS_FAMILY);
 	}
 	
 	private String getOsSpecificDownloadURL(final String os) {

--- a/src/test/java/com/kiwigrid/helm/maven/plugin/InitMojoTest.java
+++ b/src/test/java/com/kiwigrid/helm/maven/plugin/InitMojoTest.java
@@ -121,11 +121,11 @@ public class InitMojoTest {
 	private boolean isOSUnix() {
 		return System.getProperty("os.name").matches(".*n[i|u]x.*");
 	}
-	
+
 	private String getOsSpecificDownloadURL() {
 		return getOsSpecificDownloadURL(Os.OS_FAMILY == Os.FAMILY_UNIX ? "linux" : Os.OS_FAMILY);
 	}
-	
+
 	private String getOsSpecificDownloadURL(final String os) {
 		return "https://get.helm.sh/helm-v2.14.3-" + os + "-amd64." + ("windows".equals(os) ? "zip" : "tar.gz");
 	}

--- a/src/test/java/com/kiwigrid/helm/maven/plugin/InitMojoTest.java
+++ b/src/test/java/com/kiwigrid/helm/maven/plugin/InitMojoTest.java
@@ -12,6 +12,7 @@ import com.kiwigrid.helm.maven.plugin.junit.MojoExtension;
 import com.kiwigrid.helm.maven.plugin.junit.MojoProperty;
 import com.kiwigrid.helm.maven.plugin.junit.SystemPropertyExtension;
 import org.apache.maven.plugin.MojoExecutionException;
+import org.codehaus.plexus.util.Os;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
@@ -41,7 +42,7 @@ public class InitMojoTest {
 		// getHelmExecuteablePath is system-depending and has to be mocked for that reason
 		// as SystemUtils.IS_OS_WINDOWS will always return false on a *NIX system
 		doReturn(Paths.get("dummy/path/to/helm").toAbsolutePath()).when(mojo).getHelmExecuteablePath();
-		mojo.setHelmDownloadUrl("https://get.helm.sh/helm-v2.14.3-" + os + "-amd64.tar.gz");
+		mojo.setHelmDownloadUrl(getOsSpecificDownloadURL(os));
 
 		// run init
 		mojo.execute();
@@ -58,7 +59,7 @@ public class InitMojoTest {
 		// prepare execution
 		ArgumentCaptor<String> helmCommandCaptor = ArgumentCaptor.forClass(String.class);
 		doNothing().when(mojo).callCli(helmCommandCaptor.capture(), anyString(), anyBoolean());
-		mojo.setHelmDownloadUrl("https://get.helm.sh/helm-v2.14.3-linux-amd64.tar.gz");
+		mojo.setHelmDownloadUrl(getOsSpecificDownloadURL());
 
 		// run init
 		mojo.execute();
@@ -66,7 +67,7 @@ public class InitMojoTest {
 		// check captured argument
 		String helmInitCommand = helmCommandCaptor.getAllValues()
 				.stream()
-				.filter(cmd -> cmd.contains("helm init"))
+				.filter(cmd -> cmd.contains(Os.OS_FAMILY == Os.FAMILY_WINDOWS ? "helm.exe init" : "helm init"))
 				.findAny().orElseThrow(() -> new IllegalArgumentException("Only one helm init command expected"));
 
 		assertTrue(helmInitCommand.contains("--client-only"), "Option 'client-only' expected");
@@ -79,7 +80,7 @@ public class InitMojoTest {
 		// prepare execution
 		ArgumentCaptor<String> helmCommandCaptor = ArgumentCaptor.forClass(String.class);
 		doNothing().when(mojo).callCli(helmCommandCaptor.capture(), anyString(), anyBoolean());
-		mojo.setHelmDownloadUrl("https://get.helm.sh/helm-v2.14.3-linux-amd64.tar.gz");
+		mojo.setHelmDownloadUrl(getOsSpecificDownloadURL());
 		mojo.setSkipRefresh(true);
 
 		// run init
@@ -88,7 +89,7 @@ public class InitMojoTest {
 		// check captured argument
 		List<String> helmCommands = helmCommandCaptor.getAllValues()
 				.stream()
-				.filter(cmd -> cmd.contains("helm "))
+				.filter(cmd -> cmd.contains(Os.OS_FAMILY == Os.FAMILY_WINDOWS ? "helm.exe " : "helm "))
 				.collect(Collectors.toList());
 		assertEquals(1, helmCommands.size(), "Only helm init command expected");
 		String helmInitCommand = helmCommands.get(0);
@@ -97,6 +98,8 @@ public class InitMojoTest {
 
 	@Test
 	public void verifyLocalHelmBinaryUsage(InitMojo mojo) throws MojoExecutionException {
+		// Because the download URL is hardcoded to linux, only proceed if the OS is indeed linux.
+		assumeTrue(isOSUnix());
 
 		final URL resource = this.getClass().getResource("helm.tar.gz");
 		final String helmExecutableDir = new File(resource.getFile()).getParent();
@@ -112,11 +115,18 @@ public class InitMojoTest {
 		mojo.setHelmExecutableDirectory(helmExecutableDir);
 
 		// execute
-		assumeTrue(isOSUnix()); // Because the download URL is hardcoded to linux, only proceed if the OS is indeed linux.
 		mojo.execute();
 	}
 
 	private boolean isOSUnix() {
 		return System.getProperty("os.name").matches(".*n[i|u]x.*");
+	}
+	
+	private String getOsSpecificDownloadURL() {
+		return getOsSpecificDownloadURL(Os.OS_FAMILY);
+	}
+	
+	private String getOsSpecificDownloadURL(final String os) {
+		return "https://get.helm.sh/helm-v2.14.3-" + os + "-amd64." + ("windows".equals(os) ? "zip" : "tar.gz");
 	}
 }

